From fad2b9c1285a34eab0dec9f90edc06ec94e70a61 Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Tue, 1 Dec 2015 16:14:26 +0800
Subject: [PATCH] MLK-11933 mfd: mxc-hdmi-core: null check substream before
 using it

null check substream before using it. Reported by coverity.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
---
 drivers/mfd/mxc-hdmi-core.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/mfd/mxc-hdmi-core.c b/drivers/mfd/mxc-hdmi-core.c
index 76e79b7..a066b24 100644
--- a/drivers/mfd/mxc-hdmi-core.c
+++ b/drivers/mfd/mxc-hdmi-core.c
@@ -161,9 +161,12 @@ int mxc_hdmi_register_audio(struct snd_pcm_substream *substream)
 	unsigned long flags, flags1;
 	int ret = 0;
 
+	if (!substream)
+		return -EINVAL;
+
 	snd_pcm_stream_lock_irqsave(substream, flags);
 
-	if (substream && check_hdmi_state()) {
+	if (check_hdmi_state()) {
 		spin_lock_irqsave(&hdmi_audio_lock, flags1);
 		if (hdmi_audio_stream_playback) {
 			pr_err("%s unconsist hdmi auido stream!\n", __func__);
-- 
1.8.0

